### PR TITLE
ci(jenkins): enable pre-commit and some pipeline changes

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- Enable pre-commit
- tweak the log rotator for the JJBB and pipelines to keep builds for longer times
- No build branches which were filled as PRs.
- No discover tags in the downstream pipeline.

## Test cases
- pre-commit

```
git commit -m "ci(jenkins):....

Check for case conflicts.................................................Passed
Check that executables have shebangs.................(no files to check)Skipped
Check JSON...........................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Check Yaml...............................................................Passed
Check Xml............................................(no files to check)Skipped
Fix End of Files.........................................................Passed
check-bash-syntax....................................(no files to check)Skipped
check-abstract-classes-and-trait.....................(no files to check)Skipped
check-jsonslurper-class..................................................Passed
check-jenkins-pipelines..................................................Passed
check-unicode-non-breaking-spaces........................................Passed
remove-unicode-non-breaking-spaces.......................................Passed
check-en-dashes..........................................................Passed
remove-en-dashes.........................................................Passed
check-jjbb...............................................................Passed
hookid: check-jjbb

Transform JJBB to JJB
2019-08-16 09:40:26,732 [jjbb.main] [INFO] No views found at .ci/views/views.yml, continuing without them.
Validate JJB

[feature/jjbb-extend-log-rotation 03414ca] ci(jenkins): enable pre-commit in the .ci folder and tweak the log rotator for the JJBB and pipelines
 5 files changed, 31 insertions(+), 7 deletions(-)
 create mode 100644 .ci/.pre-commit-config.yaml
```

- pre-commit with a failure

```
git commit -m "chore: typo"
Check for case conflicts.................................................Passed
Check that executables have shebangs.................(no files to check)Skipped
Check JSON...........................................(no files to check)Skipped
Check for merge conflicts................................................Passed
Check Yaml...........................................(no files to check)Skipped
Check Xml............................................(no files to check)Skipped
Fix End of Files.........................................................Passed
check-bash-syntax....................................(no files to check)Skipped
check-abstract-classes-and-trait.....................(no files to check)Skipped
check-jsonslurper-class..................................................Passed
check-jenkins-pipelines..................................................Failed
hookid: check-jenkins-pipelines

Errors encountered validating Jenkinsfile:
WorkflowScript: 135: expecting '{', found 'steps' @ line 135, column 11.
             steps.junit(allowEmptyResults: true,
             ^
```

- Sanity check validation in the pipeline with a failure https://github.com/elastic/apm-agent-ruby/pull/489/commits/4f8115d5a3b2cacf53b41325e49f699d9e858e84 [here](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-ruby%2Fapm-agent-ruby-mbp/detail/PR-489/7/pipeline) show a couple of test failures: [here](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-ruby%2Fapm-agent-ruby-mbp/detail/PR-489/7/tests)

image

- Sanity check validation in the pipeline without a failure [here](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-ruby%2Fapm-agent-ruby-mbp/detail/PR-489/6/pipeline)

image

### CODE REVIEW

**Comment by mikker** [comment]
Is this something that would be run locally before committing?

File: .ci/jobs/apm-agent-ruby-mbp.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@
     description: APM Agent Ruby
     script-path: .ci/Jenkinsfile
     scm:
-    - github:
-        branch-discovery: all
+    - githu:
```
Comment (mikker) [inline .ci/jobs/apm-agent-ruby-mbp.yml]: typo?

**Comment by v1v** [comment]
> Is this something that would be run locally before committing?

Yes! You can run the below steps:
- `brew install pre-commit` or `pip install pre-commit` or other optiosn in [here](https://pre-commit.com/#install)
- then run `pre-commit install  --allow-missing-config`
- the git add ... git commit and see what's going on or run the below command to validate a file before even committing them:
```
pre-commit run --files <FILE>
```
- You can skip the pre-commit validation when running the git commit with the flag `--no-verify` or even skip a specific hook with `SKIP=hook-id git commit ....`, for instance: `SKIP=check-jjbb git commit -m 'ci(jenkins): jjbb files without any pre-commit validation`

File: .ci/jobs/apm-agent-ruby-mbp.yml (hunk 1 above)
Comment (v1v) [reply .ci/jobs/apm-agent-ruby-mbp.yml]: sorry, I didn't put a comment in the change itself but in the commit, as I do want to validate the pipeline changes as I just added a `sanity checks` stage to evaluate and detect those errors within the pipeline

Comment (v1v) [reply .ci/jobs/apm-agent-ruby-mbp.yml]: Note: this is only for testing purposes

**Comment by v1v** [comment]
WiP, as I need to validate what's going on when running the `preCommit` step as the `dockerLogin`, is somehow failing. As soon as It's solved I'll change the labels and poke you for the review.

**Comment by v1v** [comment]
and here we are! Finally, I got hit by the `refusing to merge unrelated histories` issue...  :(

**Comment by v1v** [comment]
@mikker , this is now ready to review

**Pull request merged**